### DISCUSSION TITLE
multifactor authentication

### DISCUSSION THREAD

**Discussion by robojiannis** [firecmsco/firecms#386] [Q&A] [answered]
Hi there,

I've been researching to add multifactor authentication via sms in my application. Is there such a possibility in firecms?

thanks again for the great work, the CMS is awesome!

**Comment by fgatti675** [comment, marked as answer]
Hi @robojiannis 
It is possible to use `phone` as an auth provider. I think it is what you are looking for

**Reply by robojiannis** [reply]
no, i don't want users to be able to login via phone. I want them to have to verify its them, using their phone as 2FA:
https://firebase.google.com/docs/auth/web/multi-factor

**Reply by fgatti675** [reply]
Unfortunately this is not implemented in FireCMS
PRs are welcome though!

**Reply by robojiannis** [reply]
i‘ll give it a go ;)

**Reply by robojiannis** [reply]
i got the pull request ready for you! 
I made the change in master and v1, to which branch should i push? I get a permission denied 

Definitely take a look before merging, I think there might be the firecms-way to do things here and there

**Reply by robojiannis** [reply]
this is merged for v1 and v2
